jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -144,12 +152,22 @@ def on_mouse_or_key_event(self, data):
             self.label_mouseover.reset_coords_display()
             self.label_mouseover.value = ""
 
-        elif data['event'] == 'keydown' and data['key'] == 'b':
-            self.blink_once()
+        elif data['event'] == 'keydown':
+            key_pressed = data['key']
+
+            if key_pressed == 'b':
+                self.blink_once()
+
+                # Also update the coordinates display.
+                data['event'] = 'mousemove'
+                self.on_mouse_or_key_event(data)
 
-            # Also update the coordinates display.
-            data['event'] = 'mousemove'
-            self.on_mouse_or_key_event(data)
+            elif key_pressed == 'l':
+                # Same data as mousemove above.
+                self.line_profile_xy.selected_x = data['domain']['x']
```

Comment:
@kecnry , this is not sticking in unit test that I am trying to write. The test goes through this logic route, but `lp_plugin = LineProfileXY(app=self.imviz.app)` set up in the test is not picking it up. I swear it works interactively. Do you happen to know why?

Reply:
I'm guessing `self.line_profile_xy` is not pointing to your new instance of the plugin.  This on its own could be a problem with calling plugin logic from the viewer (maybe should move to messages if we want to support multiple instances of a plugin), or you can use the same `app.get_tray_item_from_name` in the test instead of creating a new instance.

Reply:
Ah, yes, brilliant suggestion. I'll try that. thanks!